pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/Loaders.java

```diff
@@ -73,13 +73,16 @@
             indexDir.getName(), metadataVersion, configuredVersionToLoad);
       }
 
-
       // add or removes indexes based on indexLoadingConfigurationMetadata
       // add or replace new columns with default value
       // NOTE: this step may modify the segment metadata.
+      boolean isAddingDefaultColumns = false;
+      if (indexLoadingConfigMetadata != null && indexLoadingConfigMetadata.isAddingDefaultColumns()) {
+        isAddingDefaultColumns = true;
+      }
       File segmentDirectoryPath = SegmentDirectoryPaths.segmentDirectoryFor(indexDir, configuredVersionToLoad);
       try (SegmentPreProcessor preProcessor = new SegmentPreProcessor(segmentDirectoryPath, indexLoadingConfigMetadata,
-          schema)) {
+          schema, isAddingDefaultColumns)) {
```

Comment:
Should this be a part of indexLoadingConfigMetada instead of a separate parameter?